File: app/src/main/java/br/com/valenstech/letraviva/ui/home/MainActivity.java
package br.com.valenstech.letraviva.ui.home;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import br.com.valenstech.letraviva.R;
import br.com.valenstech.letraviva.databinding.ActivityMainBinding;
import br.com.valenstech.letraviva.ui.login.LoginActivity;
import br.com.valenstech.letraviva.viewmodel.AutenticacaoViewModel;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration configuracaoAppBar;
    private ActivityMainBinding binding;
    private AutenticacaoViewModel autenticacaoViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        autenticacaoViewModel = new ViewModelProvider(this).get(AutenticacaoViewModel.class);

        setSupportActionBar(binding.appBarMain.toolbar);
        DrawerLayout gaveta = binding.drawerLayout;
        NavController controladorNavegacao = Navigation.findNavController(this, R.id.nav_host_fragment);
        configuracaoAppBar = new AppBarConfiguration.Builder(
                R.id.nav_home,
                R.id.nav_devocional,
                R.id.nav_planos,
                R.id.nav_audios
        ).setOpenableLayout(gaveta).build();
        NavigationUI.setupActionBarWithNavController(this, controladorNavegacao, configuracaoAppBar);
        binding.navView.setupWithNavController(controladorNavegacao);
        binding.navView.setNavigationItemSelectedListener(menuItem -> tratarSelecaoMenu(menuItem, controladorNavegacao, gaveta));
    }

    private boolean tratarSelecaoMenu(@NonNull MenuItem itemMenu,
                                      @NonNull NavController controladorNavegacao,
                                      @NonNull DrawerLayout gaveta) {
        if (itemMenu.getItemId() == R.id.nav_logout) {
            autenticacaoViewModel.sair();
            Intent intent = new Intent(this, LoginActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(intent);
            finish();
            return true;
        }
        boolean manipulou = NavigationUI.onNavDestinationSelected(itemMenu, controladorNavegacao);
        if (manipulou) {
            gaveta.closeDrawer(GravityCompat.START);
        }
        return manipulou;
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController controladorNavegacao = Navigation.findNavController(this, R.id.nav_host_fragment);
        return NavigationUI.navigateUp(controladorNavegacao, configuracaoAppBar) || super.onSupportNavigateUp();
    }
}

File: app/src/main/java/br/com/valenstech/letraviva/ui/home/MainActivity.kt
package br.com.valenstech.letraviva.ui.home

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import br.com.valenstech.letraviva.R
import br.com.valenstech.letraviva.databinding.ActivityMainBinding
import br.com.valenstech.letraviva.viewmodel.AuthViewModel
import br.com.valenstech.letraviva.ui.login.LoginActivity

class MainActivity : AppCompatActivity() {

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding
    private lateinit var authViewModel: AuthViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        authViewModel = ViewModelProvider(this)[AuthViewModel::class.java]

        setSupportActionBar(binding.appBarMain.toolbar)
        val drawerLayout: DrawerLayout = binding.drawerLayout
        val navController = findNavController(R.id.nav_host_fragment)
        appBarConfiguration = AppBarConfiguration(
            setOf(R.id.nav_home, R.id.nav_devocional, R.id.nav_planos, R.id.nav_audios, R.id.nav_settings),
            drawerLayout
        )
        setupActionBarWithNavController(navController, appBarConfiguration)
        binding.navView.setupWithNavController(navController)
        binding.navView.setNavigationItemSelectedListener { menuItem ->
            if (menuItem.itemId == R.id.nav_logout) {
                authViewModel.signOut()
                val intent = Intent(this, LoginActivity::class.java).apply {
                    flags =
                        Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
                }
                startActivity(intent)
                finish()
                true
            } else {
                val handled = NavigationUI.onNavDestinationSelected(menuItem, navController)
                if (handled) {
                    drawerLayout.closeDrawer(GravityCompat.START)
                }
                handled
            }
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment)
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }
}
